.github/workflows/docker-image.yml

```diff
@@ -6,22 +6,30 @@ on:
     branches: [ "main" ]
   
 jobs:
-  login:
+  build_and_publish:
     runs-on: ubuntu-latest
     steps:
-    # Login in to ghcr
-    - name: Login to GitHub Container Registry
-      uses: docker/login-action@v2
-      with:
+      # Log in to ghcr
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
           registry: ghcr.io
           username: ${{ github.repository }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      
+
+      # Checkout
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      # Set version
+      - name: Set version
+        id: version
+        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
       # Build the image with tag matching repo owner name
-    - uses: actions/checkout@v3
-    - name: Build the Docker image
-      run: docker build -t ghcr.io/${{ github.repository }} .
+      - name: Build the Docker image
+        run: docker build --build-arg VERSION=${{ steps.version.outputs.sha_short }} -t ghcr.io/${{ github.repository }} .

```

Comment:
I was thinking of using a more sequential number like 1.0.0 then 1.0.1 as we go deeper, what does a45a2b4 do for us? Looking at the code how do you know we are using version a45a2b4?

Reply:
like {{github.run_id}} so we can can just look at the pipelines and see the latest version deploy

Reply:
Sure, that is simpler. It should still be shortened to 7 chars when the bot sends it, though. 40 is too long

Reply:
hmm, it makes practical sense, for me personally I just like simpler things, so later down the road we aren't looking at this command trying to figure out whats going on, but we can push if you want